piptools/resolver.py

```diff
@@ -448,3 +478,164 @@ def reverse_dependencies(
             ireq for ireq in ireqs if not (ireq.editable or is_url_requirement(ireq))
         ]
         return self.dependency_cache.reverse_dependencies(non_editable)
+
+
+class Resolver(BaseResolver):
+    """
+    Uses pip's 2020 resolver.
+    """
+
+    def __init__(
+        self,
+        constraints: Iterable[InstallRequirement],
+        existing_constraints: typing.Dict[str, InstallRequirement],
+        repository: BaseRepository,
+        allow_unsafe: bool = False,
+        **kwargs: Any,
+    ) -> None:
+        self.constraints = list(constraints)
+        self.repository = repository
+        self.allow_unsafe = allow_unsafe
+
+        options = self.options = self.repository.options
+        self.session = self.repository.session
+        self.finder = self.repository.finder
+        self.command = self.repository.command
+        self.unsafe_constraints: Set[InstallRequirement] = set()
+
+        self.existing_constraints = existing_constraints
+        self._constraints_map = {key_from_ireq(ireq): ireq for ireq in constraints}
+
+        if "2020-resolver" not in options.features_enabled:
+            raise PipToolsError("2020 resolver feature must be enabled.")
+
+        # Make sure there is no enabled legacy resolver
+        options.deprecated_features_enabled = remove_value(
+            options.deprecated_features_enabled, "legacy-resolver"
+        )
+
+    def resolve(self, max_rounds: int = 10) -> Set[InstallRequirement]:
+        with get_requirement_tracker() as req_tracker, global_tempdir_manager(), indent_log(), update_env_context_manager(  # noqa: E501
```

Comment:
Why don't you just fix the line length instead of ignoring it? You could reorder the CMs to put the update env first (unless it's order-sensitive).
Also, https://pypi.org/project/flake8-length/ exists. It could allow occasional length violations.

Reply:
Black can't format long `with` statements. See https://github.com/psf/black/issues/664

Reply:
Addressed in ac2f5cb24b5bbd4da06fe1c8a43b362aa3c0b868